### ISSUE TITLE
JEID Not increasing ID limit for biomes

### ISSUE THREAD

**Issue by AlmightyChan** [DimensionalDevelopment/JustEnoughIDs#111]
Having an issue where anytime I try to load or create an OTG World with BoP or Normal causes crash below is what catches my  #attention

java.lang.RuntimeException: Could not allocate the requested id 256 for biome mapleriverlands. All available id's under 256 have been allocated
. To proceed, adjust your WorldConfig or use the ReplaceToBiomeName feature to make the biome virtual.

 #101 

[crash-2020-01-14_01.57.56-server.txt](https://github.com/DimensionalDevelopment/JustEnoughIDs/files/4058659/crash-2020-01-14_01.57.56-server.txt)

**Comment by sam-kirby** [comment]
Both OTG and JEID make significant changes to how biome registration works. JEID's fixes will not work in combination with OTG as they are bypassed.

Duplicate of #101

**Comment by Dread131313** [comment]
So we basically have to choose to either use OTG with its presets which add many biome IDs WITHOUT JEID, and risk running into issues with other non OTG mods (like advanced rocketry) and hitting the biome limit that way (unless we use lesser conventional mods with lesser biome count)...

*OR*

NOT use OTG, but instead use JEID with conventional mods (or at the very least use OTG with this, but with presets that have a low biome ID count), but the two wont work together as they basically cancel each other out?

The problem I am having is I have JEID installed along with OTG, as long as I use presets that dont add lots of biome ID's its fine, but if I load up something like BiomeBundle or Travelers Dream (both of which add upwards of 600 biomes EACH, which is somewhat ridiculous), I get "cannot load dimension due to not enough biome ID's". 

I looked in the BiomeID registry from MultiMob and it says I am registering 142 Biome ID's from various mods (abyssalcraft, betweenlands, midnight...etc). When I add dimensions/presets as worlds into my world with presets like travelers dream or biome bundle is when I have issues.

So it looks like both mods (OTG and JEID) are more or less cancelling each other out. or at the very least one is working but the other isnt because of a conflict as sam-kirby stated. I sort of feel like the OTG part is  being cancelled out so when OTG presets try to add biome ID's it just gets lumped into the vanilla batch, which the ID limit is 256. Actually now that I think about it, its almost as if both are not working and I have to choose one or the other.

At this point im going to try and see if i can revert back to NEI and forgo JEID, ultimately I might just have to bite the bullet and opt out of not using OTG presets that have a large biome count - which is fine, its what ive been doing up till now, was just trying to see if there is some workaround or solution to this - but I sort of feel like I answered my own question with this post so... thanks? :)

TLDR: Choose one or the other, OTG without JEID installed with big presets but watch out for using biome IDs otherwise

*OR*

Do not use OTG (or only use low biome ID count presets) along with JEID (if possible??) and have lower biome counts for OTG presets.

**Comment by Dread131313** [comment]
Update: So I found a workaround to this issue. What I did was uninstall JEID, when I did this, it created another issue - not enough ID's so I downloaded and installed NEI (not enough IDs). This still posed a problem with potion ID's since mojang set the bar really low and I had a mod (Wards) adding many potion IDs since those are based off of the amount of enchantments loaded (to which there were many and since NEI increases enchantment ID but not potion ID it was a lot), I opted to remove potioncore and qualitytools but this didnt work, so I instead downloade Potion ID extender, and this resolved the potion ID issue.

Once I did all of this I was able to load into minecraft fully with 355 mods and (approx 146 basic modded biome IDs loaded) and was able to load up all of the OTG presets with all the biomes without issue. Hope someone reading this is helped.

TLDR: Remove JEID (if using OTG and if when using OTG you are using presets with large biome ID count) and install NEI as well as Potion ID extender and you shou;d be fine.

Personal thoughts: JEID is a great mod, the fact I am using NEI in lieu of JEID doesnt mean JEID is a bad mod, only the limitations of it working with OTG (which is arguably because both OTG and JEID manipulate biome IDs, which causes a conflict) doesnt work, so my case - is very rare and unique probably since admittedly most people will either just use OTG without needing JEID or use JEID without ever touching OTG. Since I am building a "kitchen-sink" modpack and I utterly detest biomes'o'plenty, this was the best course of action for me.

I used JEID for many years, as it got me out of a jam with not enough potion IDs back in the day, so its still a great mod, just cannot use it with OTG+big presets is all.

**Comment by CitrusHappy** [comment]
Anyone else having this issue, check the JEID discord server in announcements. The OTG and JEID team have been working on a modified version of OTG to work with both the "virtual" and real biomes at the same time. So you can have a large modpack and use a large OTG preset at the same time.

**Comment by matyklug18** [comment]
> Anyone else having this issue, check the JEID discord server in announcements. The OTG and JEID team have been working on a modified version of OTG to work with both the "virtual" and real biomes at the same time. So you can have a large modpack and use a large OTG preset at the same time. I've tested it in my personal pack with this issue and it has been fixed.

What is the current status of this?
